proof/refine/RISCV64/Invariants_H.thy

```diff
@@ -130,29 +167,151 @@ lemma st_tcb_at'_def:
   "st_tcb_at' test \<equiv> obj_at' (test \<circ> tcbState)"
   by (simp add: pred_tcb_at'_def o_def)
 
+definition active_sc_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
+  "active_sc_at' \<equiv> obj_at' (\<lambda>ko :: sched_context. 0 < scRefillMax ko)"
+
 text \<open> cte with property at \<close>
 definition cte_wp_at' :: "(cte \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
   "cte_wp_at' P p s \<equiv> \<exists>cte::cte. fst (getObject p s) = {(cte,s)} \<and> P cte"
 
 abbreviation cte_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
   "cte_at' \<equiv> cte_wp_at' \<top>"
 
+text \<open>replyNext aliases\<close>
+
+abbreviation replySC :: "reply \<Rightarrow> obj_ref option" where
+  "replySC \<equiv> \<lambda>r. getHeadScPtr (replyNext r)"
+
+abbreviation replyNext_of :: "reply \<Rightarrow> obj_ref option" where
+  "replyNext_of \<equiv> \<lambda>r. getReplyNextPtr (replyNext r)"
+
+lemma getReplyNextPtr_None[simp]:
+  "getReplyNextPtr None = None" by (simp add: getReplyNextPtr_def)
+
+lemma getHeadScPtr_None[simp]:
+  "getHeadScPtr None = None" by (simp add: getHeadScPtr_def)
+
+lemma getReplyNextPtr_Some_Next[simp]:
+  "getReplyNextPtr (Some (Next rn)) = Some rn" by (simp add: getReplyNextPtr_def)
+
+lemma getHeadScPtr_Some_Head[simp]:
+  "getHeadScPtr (Some (Head sc)) = Some sc" by (simp add: getHeadScPtr_def)
+
+lemma theReplyNextPtr_Some_Next[simp]:
+  "theReplyNextPtr (Some (Next rn)) = rn" by (simp add: theReplyNextPtr_def)
+
+lemma theHeadScPtr_Some_Head[simp]:
+  "theHeadScPtr (Some (Head sc)) = sc" by (simp add: theHeadScPtr_def)
```

Comment:
these should be replaced with lemmas of the form

`lemmas getReplyNextPr_simps[simp] = getReplyNextPtr_def[split_simps option.split reply_next.split]`

(This will cover all cases and make it resistant to change in constructor names and pattern)

Reply:
I'm pretty sure I said the same on the original PR where this was introduced, but apparently that has not made it in.